### PR TITLE
Remove obsolete members for v7 (aka v6)

### PR DESCRIPTION
**Pull request by Shane32**

### CODE REVIEW

File: docs2/site/docs/migrations/migration6.md [hunk 1]
```diff
@@ -13,6 +13,34 @@ memory usage is reduced by freeing unnecessary references after obtaining the re
...
+This method adds the `InstrumentFieldsMiddleware` to the schema, and conditionally enables metrics
+during execution via `ExecutionOptions.EnableMetrics`. It also appends the Apollo Tracing results
+to the execution result if metrics is enabled during execution.
+
+### 4. Add `ConfigureExecution` builder method (added in 5.3.0)
+
+`ConfigureExecution` allows a delegate to both alter the execution options and the execution result.
+For example, to add total execution time to the results, you could write:
+
+```cs
```
Comment (sungam3r) [inline docs2/site/docs/migrations/migration6.md:27]: csharp

File: docs2/site/docs/migrations/migration6.md [hunk 2]
```diff
@@ -13,6 +13,34 @@ memory usage is reduced by freeing unnecessary references after obtaining the re
 
 Particularly useful for authentication checks, now validation rules are asynchronous.
 
+### 3. Add `AddApolloTracing` builder method (added in 5.3.0)
+
+This method adds the `InstrumentFieldsMiddleware` to the schema, and conditionally enables metrics
+during execution via `ExecutionOptions.EnableMetrics`. It also appends the Apollo Tracing results
+to the execution result if metrics is enabled during execution.
+
+### 4. Add `ConfigureExecution` builder method (added in 5.3.0)
```
Comment (sungam3r) [inline docs2/site/docs/migrations/migration6.md:22]: Why here for v6 migration?

File: src/GraphQL/DI/IConfigureExecutionOptions.cs [hunk 3]
```diff
@@ -29,41 +14,18 @@ public ConfigureExecutionOptions(Action<ExecutionOptions> action)
...
-        public Task ConfigureAsync(ExecutionOptions executionOptions)
+        public Task<ExecutionResult> ExecuteAsync(ExecutionOptions options, ExecutionDelegate next)
         {
-            return _action(executionOptions);
-        }
-    }
+            var task = _action(options);
+            if (task.IsCompleted && task.Status == TaskStatus.RanToCompletion)
+                return next(options);
+            return Continuer(task, options, next);
```
Comment (sungam3r) [inline src/GraphQL/DI/IConfigureExecutionOptions.cs:22]: So you want to gain a little perf here, right? You could just always await here, it's easier. I'm fine either way.

File: docs2/site/docs/migrations/migration6.md [hunk 4]
```diff
@@ -24,3 +52,44 @@ Removal was necessary as the value is released after the result is set.
...
+Please consider the operation of the new `AddApolloTracing` method (see 'New Features' section above)
+when replacing `AddMetrics` with `AddApolloTracing`. Remember that `AddApolloTracing` includes
+functionality previously within `ApolloTracingDocumentExecuter` and/or `EnrichWithApolloTracing`.
+
+### 3. `GlobalSwitches.MapAllEnumerableTypes` has been removed; only specific types are detected as lists.
+
+When auto detecting graph types from CLR types (usually within `AutoRegisteringObjectGraphType` or the
+expression syntax of `Field(x => x.Member)`), previously any type except `string` that implemented
+`IEnumerable` was considered a list type. This would includes types such as dictionary types, making
+it impossible to register a CLR type that dervies from a dictionary for automatic mapping.
```
Comment (sungam3r) [inline docs2/site/docs/migrations/migration6.md:80]:
```suggestion
it impossible to register a CLR type that derives from a dictionary for automatic mapping.
```

**Review by sungam3r — APPROVED** [review]

File: docs2/site/docs/migrations/migration6.md (hunk 2 above)
Comment (Shane32) [reply docs2/site/docs/migrations/migration6.md:22]: It is not listed in any other migration notes, and is worth mentioning.

File: src/GraphQL/DI/IConfigureExecutionOptions.cs (hunk 3 above)
Comment (Shane32) [reply src/GraphQL/DI/IConfigureExecutionOptions.cs:22]: I would much rather remove this condition and duplicate code:

https://github.com/graphql-dotnet/graphql-dotnet/blob/master/src/GraphQL/Validation/DocumentValidator.cs#L86

Comment (sungam3r) [reply src/GraphQL/DI/IConfigureExecutionOptions.cs:22]: 😄

**Pull request merged**